thresher/models.py

```diff
@@ -88,11 +88,17 @@ class Topic(models.Model):
     # an id within the given Analysis Type
     topic_id = models.IntegerField() 
 
-    # The analysis type to which this topic belongs
-    analysis_type = models.ForeignKey(AnalysisType, related_name='topics')
+    # an id of its parent topic
+    parent_id = models.ForeignKey(self, related_name="subtopics")
```

Comment:
Have you tried running this? I think this might throw an error because self is not defined in this context (these are class variables and there is no instance to refer to). I think this should be replaced by 'self' as a string according to the docs: https://docs.djangoproject.com/en/dev/ref/models/fields/#foreignkey.

Also, not as important, but you might want to set the on_delete argument to models.CASCADE. It's the current default but in newer versions of Django it will raise a warning and will completely fail as of Django 2.0.